src/rules/order.js

```diff
@@ -310,24 +310,29 @@ function computePathRank(ranks, pathGroups, path, maxPosition) {
   }
 }
 
-function computeRank(context, ranks, name, type, excludedImportTypes) {
-  const impType = importType(name, context)
+function computeRank(context, node, ranks, name, type, excludedImportTypes) {
+  let impType
+  if (type === 'import:object') {
+    impType = 'object'
+  } else {
+    impType = importType(name, context)
+  }
   let rank
   if (!excludedImportTypes.has(impType)) {
     rank = computePathRank(ranks.groups, ranks.pathGroups, name, ranks.maxPosition)
   }
   if (typeof rank === 'undefined') {
     rank = ranks.groups[impType]
   }
-  if (type !== 'import') {
+  if (!/import(:.*)?/.test(type)) {
```

Comment:
```suggestion
  if (!type.startsWith('import')) {
```

Reply:
When using "startsWith" it's not ensured, that it's either just "import" or "import:something". Is this change meant for performance gain?

Reply:
I'd recommend to change this to `^import(:.*)?$/.test(type)` to be sure, that `type` either matches `import` or `import:something`.

Reply:
I'd prefer a non-regex check - even multiple `===` would be better.